src/schemas/components/SchemaFormPropertyDate.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <p-date-input v-model="value" clearable />
+</template>
+
+<script lang="ts" setup>
+  import { isNotNullish } from '@prefecthq/prefect-design'
+  import { format, parse, startOfToday } from 'date-fns'
+  import { computed } from 'vue'
+  import { isInvalidDate } from '@/utilities'
+
+  const props = defineProps<{
+    value: string | null,
+  }>()
+
+  const emit = defineEmits<{
+    'update:value': [string | null],
+  }>()
+
+  const dateFormat = 'yyyy-MM-dd'
+
+  const value = computed({
+    get() {
+      if (isNotNullish(props.value)) {
+        const parsed = parse(props.value, dateFormat, startOfToday())
+
+        if (isInvalidDate(parsed)) {
+          return null
+        }
```

Comment:
For this and datetime properties - does a partial date throw null here? Thinking through the experience as someone types a date in manually instead of using the picker

Reply:
The component doesn't emit until there is a valid date I believe. So this is only a safe guard for a value that already exists as a string that cannot be parsed.